### PR TITLE
Handle pep files with `.rst` extensions.

### PR DESCRIPTION
**Pull request by Carreau**
Not completely sure how that would interact with deploying pep on pythondotorg and other services.

### CODE REVIEW

File: pep2html.py [hunk 1]
```diff
@@ -469,7 +471,7 @@ def pep_type_error(inpath, pep_type):
 def browse_file(pep):
     import webbrowser
     file = find_pep(pep)
-    if file.endswith(".txt"):
+    if (file.startswith('pep-') and (file.endswith(".txt") or file.endswith('.rst'))):
```
Comment (berkerpeksag) [inline pep2html.py]: `file.endswith((".txt", ".rst"))` would be clearer.

File: pep2html.py [hunk 2]
```diff
@@ -478,7 +480,7 @@ def browse_file(pep):
 def browse_remote(pep):
     import webbrowser
     file = find_pep(pep)
-    if file.endswith(".txt"):
+    if (file.startswith('pep-') and (file.endswith(".txt") or file.endswith('.rst'))):
```
Comment (berkerpeksag) [inline pep2html.py]: `file.endswith((".txt", ".rst"))` would be clearer.

**Comment by berkerpeksag** [comment]
> Not completely sure how that would interact with deploying pep on pythondotorg and other services.

I don't have time to test it this week, but it looks like a backward compatible change to me.

Thanks!

**Comment by Carreau** [comment]
The only undefined behavior is if txt and rst are present.

I'll update with your suggestions later.

**Comment by Carreau** [comment]
Updated as per comment.

**Comment by Carreau** [comment]
You can see the difference in the generated html [here](https://github.com/python/peps/compare/c03a521ed103defbd1b5da50170cfb81105b6e59...Carreau:regen-html) , which is mostly due to the `random.choice` of banner, and (strangely) a couple of author that are not sorted in the same order in pep0. (investigating, but might be unrelated to that).

File: genpepindex.py [hunk 3]
```diff
@@ -41,7 +41,7 @@ def main(argv):
             abs_file_path = os.path.join(path, file_path)
             if not os.path.isfile(abs_file_path):
                 continue
-            if file_path.startswith("pep-") and file_path.endswith(".txt"):
+            if file_path.startswith("pep-") and file_path[-4:] in (".txt", "rst"):
```
Comment (berkerpeksag) [inline genpepindex.py]: `file_path.endswith((".txt", ".rst"))` would work too.

File: pep2html.py [hunk 4]
```diff
@@ -367,10 +366,13 @@ def get_input_lines(inpath):
...
 

 def find_pep(pep_str):
-    """Find the .txt file indicated by a cmd line argument"""
+    """Find the .rst or .txt file indicated by a cmd line argument"""
     if os.path.exists(pep_str):
         return pep_str
     num = int(pep_str)
+    rstpath = "pep-%04d.rst"
+    if os.path.exists(rstpath):
```
Comment (berkerpeksag) [inline pep2html.py]: I think you forgot to pass `num` to `rstpath` here.

File: pep2html.py [hunk 5]
```diff
@@ -520,32 +522,32 @@ def main(argv=None):
             browse = 1
 
     if args:
-        peptxt = []
+        peptxtrst = []
```
Comment (berkerpeksag) [inline pep2html.py]: I'd rename it to `peps`, `pep_list` or something else (`peptxtrst` won't make sense once we convert all peps to reST).

**Comment by ncoghlan** [comment]
Big +1 for the general idea, since setting the file extension will let GitHub render them nicely.

At the detailed level, I didn't have anything to add to Berker's existing comments.

**Comment by ncoghlan** [comment]
That said, would it make sense to update https://www.python.org/dev/peps/pep-0012/ at the same time as making the changes to support the .rst extension?

It turns out that needs an update anyway to refer to the GitHub repo rather than hg.python.org

**Comment by Carreau** [comment]
Will do both early next week.

File: pep2html.py (hunk 5 above)
Comment (IanLee1521) [inline pep2html.py]: Personally I prefer `pep_list`.

**Comment by Carreau** [comment]
Updated as well as pep12.

**Comment by brettcannon** [comment]
Thanks for this, @Carreau !

**Pull request merged**